### PR TITLE
Add integration tests for SAML-OIDC federation.

### PR DESCRIPTION
**Pull request by ChamathNS**
Resolves https://github.com/wso2/product-is/issues/10578.
The integration tests include the common scenarios, "Federated Login" and "Federated Logout".

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+        Assert.assertNotNull(decodedSAMLResponse);
+
+        String logoutPageContent = sendSAMLResponseToWebApp(samlLogoutResponseToWebapp);
+        boolean isValidLogout = validateLogoutPageContent(logoutPageContent);
+        Assert.assertTrue(isValidLogout, "Invalid SAML Logout response received by travelocity app");
+    }
+
+    //TODO Test case for consent denial from the federated IdP during the logout.
+//    @Test(groups = "wso2.is", description = "Check SAML-to-OIDC federated logout deny-consent", dependsOnMethods = {
+//            "testFederatedLogin"})
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java:198]: Why this is commented?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+            IOException {
+
+        HttpGet getRequest = new HttpGet(locationURL);
+        getRequest.addHeader("User-Agent", OAuth2Constant.USER_AGENT);
+        HttpResponse response = client.execute(getRequest);
+
+        return response;
+    }
+
+    public HttpResponse sendLogoutApprovalPostWithConsent(HttpClient client) throws IOException {
```
Comment (ShanChathusanda93) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java:724]: Shall we make the method descriptions in order to maintain the format. WDYT?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,760 @@
...
+        request.setEntity(new UrlEncodedFormEntity(urlParameters));
+
+        HttpResponse response = client.execute(request);
+        return response;
+    }
+
+    /**
+     * Send Get request
+     *
+     * @param locationURL - Get url location
```
Comment (ShanChathusanda93) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java:709]: Shall we add some information to this @param descriptions. WDYT?

**Comment by piraveena** [comment]
@ChamathNS this OIDCIdentityFederationTestCase is not added to the testng.xml file

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java (hunk 1 above)
Comment (emswbandara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java:198]: let's add the reason for making this as todo in the comments and link to the git issue if possible

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 4]
```diff
@@ -0,0 +1,760 @@
...
+import org.opensaml.xml.util.Base64;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.JustInTimeProvisioningConfig;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.Property;
+import org.wso2.carbon.identity.application.common.model.xsd.*;
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java:43]: remove * imports

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 5]
```diff
@@ -0,0 +1,760 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.rmi.RemoteException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class OIDCIdentityFederationTestCase extends AbstractIdentityFederationTestCase {
```
Comment (emswbandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java:65]: add class level comment

Comment (ChamathNS) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java:74]: Addressed with https://github.com/wso2/product-is/pull/10640/commits/1229a0d9d2cb9bb365a6c0aec4236198bd667a2c.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java (hunk 4 above)
Comment (ChamathNS) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: Addressed with https://github.com/wso2/product-is/pull/10640/commits/1229a0d9d2cb9bb365a6c0aec4236198bd667a2c.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java (hunk 3 above)
Comment (ChamathNS) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: Addressed with https://github.com/wso2/product-is/pull/10640/commits/1229a0d9d2cb9bb365a6c0aec4236198bd667a2c.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java (hunk 2 above)
Comment (ChamathNS) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: Addressed with https://github.com/wso2/product-is/pull/10640/commits/1229a0d9d2cb9bb365a6c0aec4236198bd667a2c.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java (hunk 1 above)
Comment (ChamathNS) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java:209]: Addressed with https://github.com/wso2/product-is/pull/10640/commits/1229a0d9d2cb9bb365a6c0aec4236198bd667a2c.

**Comment by ChamathNS** [comment]
> @ChamathNS this OIDCIdentityFederationTestCase is not added to the testng.xml file in this PR

Addressed with https://github.com/wso2/product-is/pull/10640/commits/1229a0d9d2cb9bb365a6c0aec4236198bd667a2c.

**Review by piraveena — APPROVED** [review]

**Comment by ChamathNS** [comment]
Merged after running the test case locally and verifying it completed successfully.

**Pull request merged**